controllers/updateservice_controller.go

```diff
@@ -516,16 +516,36 @@ func validateRouteName(instance *cv1.UpdateService, name string, namespace strin
 }
 
 func (r *UpdateServiceReconciler) ensurePolicyEngineRoute(ctx context.Context, reqLogger logr.Logger, instance *cv1.UpdateService, resources *kubeResources) error {
+	oldRoute := resources.policyEngineOldRoute
 	route := resources.policyEngineRoute
+
 	// Set UpdateService instance as the owner and controller
 	if err := controllerutil.SetControllerReference(instance, route, r.Scheme); err != nil {
 		return err
 	}
```

Comment:
Shouldn't we wait to do this until we know which route, old or new, we're going to use?